### DISCUSSION TITLE
Is Xcode on MacOS really necessary?

### DISCUSSION THREAD

**Discussion by avalon60** [TomSchimansky/CustomTkinter#1389] [Q&A]
Apologies, if details are a little vague here, as I don't have a Mac. 

I recently wrote an CustomTkinter based app, which runs on Windows, Linux and MacOS. However, there was a slight complication with Mac. For some reason the guy who tested the program for me, had to perform the following steps to get it working:

```
	brew install libxml2
	brew install libxslt
	brew install python
	xcode-select --install
	brew install python-tk
```

Here is the Python requirements file contents:

```
bcrypt==4.0.1
certifi==2022.9.24
cffi==1.15.1
circuitbreaker==1.4.0
cryptography==38.0.4
customtkinter==4.6.3
darkdetect==0.7.1
kellanb-cryptography==1.2.2
lxml==4.7.1
oci==2.88.2
Pillow==9.3.0
pycparser==2.21
pycryptodome==3.15.0
pyOpenSSL==22.0.0
python-dateutil==2.8.2
pytz==2022.6
six==1.16.0
tkinter-tooltip==2.1.0
```

Is this normal on a Mac? 

Thanks in advance,

Clive

**Comment by Akascape** [comment]
XCode is not necessary, you can just install python only (package can be downloaded from python.org).

**Reply by avalon60** [reply]
Thanks Akash. I thought it odd.
